### ISSUE TITLE
Each HOST can have multiple certs. Each REQUEST can select which cert to use.

### ISSUE THREAD

**Issue by stevenvandyke** [a85/POSTMan-Chrome-Extension#11150]
### Is there an existing request for this feature?

- [X] I have searched the existing issues for this feature request and I know that duplicates will be closed

### Is your feature request related to a problem?

Our API has one host "abc.example.com" shared by multiple controllers containing many endpoints. The problem is, some of our endpoints require an Admin Cert, some require a User Cert others required a Device Cert, etc. So to test all of our endpoints we have to keep removing and adding the ONE cert allowed for our host.

### Describe the solution you'd like

The simple solution would be to allow users to add multiple certificates to each host and let them select the default certificate they want to use (toggling is much easier than removing and adding).

    Host: qa.myhost.com
    [X] PFX file:   path to my qa ADMIN cert
        Passphrase: ******
    [ ] PFX file:   path to my qa USER cert
        Passphrase: ******
    [ ] PFX file:   path to my qa DEVICE cert
        Passphrase: ******

Another enhancement would be to allow users to specify the certificate index for each request (eliminating the need to open the Certificates window at all).

    Request A > Settings
        Certificate index = Default
    Request B > Settings
        Certificate index = 1
    Request C > Settings
        Certificate index = 2

### Describe alternatives you've considered

This is about the simplest design I could come up with considering the current design of the UI and the relationships between Requests, Certificates and Environments.

**Comment by DannyDainton** [comment]
Hey @stevenvandyke 

Is this feature request similar to the one you're asking for here?

https://github.com/postmanlabs/postman-app-support/issues/4033

**Comment by stevenvandyke** [comment]
Yes, it is the same basic problem we are both reporting. His #2 suggestion is like mine, but my suggestion goes into a bit more detail. I wish you could make this a high priority, as I'm sure lots of users are just dealing with the pains of the current design. Thank you.

-Steve
________________________________

**Comment by sulthompson** [comment]
+1 This feature would be very useful.  It's also similar to #4753

**Comment by cspielmann** [comment]
+1 This feature would be very useful.

**Comment by ejvk** [comment]
+1 This would be so helpful and save a lot of time but not having to constantly change the certificates

**Comment by sundarpn** [comment]
+1 
We have web service that use client cert auth. I too have a need to use different kind of certificates for the same host/fqdn. 

e.g. foo.something.com/service 
- normal user would auth to this presenting one kind of certificate.  
- while admin users would present a different certificate.  

Today, I have to painfully go into Postman > settings > certificates and reinstall different certificates (.pfx). This prevents any true automated test.

**Comment by cdelanchy** [comment]
A +1 here too.

I have a reverse-proxy which sends different requests to different services. These services are accessible with the subdomain (abc.test.fr, and xyz.test.fr), so sharing the same domain.

Each service uses a mTLS connection with a different certs. And for testing and QA, I can't register 2 client certificates for the FQDN (one for abc.test.fr and one for xyz.test.fr) and Postman choose the cert' accordingly, because Postman only choose the cert with the domain name (test.fr) and pick the first one.

**Comment by CS-Roche** [comment]
+1 from me. We have a huge API and each endpoint is authorized for multiple users. Each user has their own certificate. As you can imagine, it becomes SUPER cumbersome to constantly switch around certificates when doing testing. Go to settings, click around file system to select folder containing certificates, choose certificate, open password manager, copy password, paste it in the password field, click OK. Over and over and over and over.

Good times.

**Comment by simi1029** [comment]
+1 for this

**Comment by giulianabaratto** [comment]
+1

**Comment by yaronuliel** [comment]
+1
This is such a basic feature for working with API Gateways that have multiple services on the same domain
It's amazing that this request is arround for over 7 years ([#4033](https://github.com/postmanlabs/postman-app-support/issues/4033))! with no solution, while every other alternative tool that support mTLS have it.

**Comment by jonlindy** [comment]
+1
2026 and this is still an issue. Truly incredible support
We're all crying into the void at this point
